packages/core/src/new-api/internal/new-execution/type-guards/execution-state.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  CallExecutionState,
+  ContractAtExecutionState,
+  DeploymentExecutionState,
+  ExecutionSateType,
+  ExecutionState,
+  SendDataExecutionState,
+  StaticCallExecutionState,
+} from "../types/execution-state";
+
+export function isExecutionState(
+  potential: unknown
+): potential is DeploymentExecutionState {
+  return (
+    isDeploymentExecutionState(potential) ||
+    isCallExecutionState(potential) ||
+    isStaticCallExecutionState(potential) ||
+    isContractAtExecutionState(potential) ||
+    isReadEventArgumentExecutionState(potential) ||
+    isSendDataExecutionState(potential)
+  );
+}
+
+export function isDeploymentExecutionState(
+  potential: unknown
```

Comment:
Thinking about every typeguard as taking `unknown` may seem to indicate that they are needed. But this one, for example, is only used in context where we already know we have an execution state, so all we need to do is using its `type`.